src/main/java/redis/clients/jedis/SslOptions.java

```diff
@@ -0,0 +1,502 @@
+            this.sslParameters = sslParameters;
+            return this;
+        }
+
+        public Builder sslVerifyMode(SslVerifyMode sslVerifyMode) {
+            this.sslVerifyMode = sslVerifyMode;
+            return this;
+        }
+
+        public Builder sslContextProtocol(String protocol) {
+            this.sslContextProtocol = protocol;
+            return this;
+        }
+
+        /**
+         * Create a new instance of {@link SslOptions}
+         *
+         * @return new instance of {@link SslOptions}
+         */
+        public SslOptions build() {
+            if (this.sslParameters == null) {
+                this.sslParameters = new SSLParameters();
+            }
+            return new SslOptions(this);
+        }
+
+    }
+
+    /**
+     * A {@link SSLContext} object that is configured with values from this {@link SslOptions} object.
+     *
+     * @return {@link SSLContext}
+     * @throws IOException thrown when loading the keystore or the truststore fails.
+     * @throws GeneralSecurityException thrown when loading the keystore or the truststore fails.
+     */
+    public SSLContext createSslContext() throws IOException, GeneralSecurityException {
+
+        TrustManager[] trustManagers = null;
+
+        if (sslVerifyMode == SslVerifyMode.FULL) {
+            this.sslParameters.setEndpointIdentificationAlgorithm("HTTPS");
+        } else if (sslVerifyMode == SslVerifyMode.CA) {
+            this.sslParameters.setEndpointIdentificationAlgorithm("");
+        } else if (sslVerifyMode == SslVerifyMode.INSECURE) {
+            trustManagers = new TrustManager[] { INSECURE_TRUST_MANAGER };
+        }
+
+        KeyManager[] keyManagers = null;
+        if (keystoreResource != null) {
+
+            KeyStore keyStore = KeyStore.getInstance(keyStoreType);
+            try (InputStream keystoreStream = keystoreResource.get()) {
+                keyStore.load(keystoreStream, keystorePassword);
+            }
+
+            KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(keyManagerAlgorithm);
+            keyManagerFactory.init(keyStore, keystorePassword);
+            keyManagers = keyManagerFactory.getKeyManagers();
+        }
+
+        if (trustManagers != null) {
+            // already processed
+        } else if (truststoreResource != null) {
+
+            KeyStore trustStore = KeyStore.getInstance(trustStoreType);
+            try (InputStream truststoreStream = truststoreResource.get()) {
+                trustStore.load(truststoreStream, truststorePassword);
+            }
+
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(trustManagerAlgorithm);
+            trustManagerFactory.init(trustStore);
+            trustManagers = trustManagerFactory.getTrustManagers();
+        }
+
+        SSLContext sslContext = SSLContext.getInstance(sslContextProtocol);
+        sslContext.init(keyManagers, trustManagers, null);
```

Comment:
About `null` SecureRandom: https://stackoverflow.com/questions/295628/securerandom-init-once-or-every-time-it-is-needed